### PR TITLE
Configuration authorization: Permission checks and editing of permissions for users and groups

### PR DESCRIPTION
**Pull request by StefanFl**
This PR contains the foundation to edit and check permissions for configurations. When completed, we have a flexible way to deal with permissions for everything that's not under a product type or product and can remove the `is_staff` flag.

There is a feature flag `FEATURE_CONFIGURATION_AUTHORIZATION` to be able to switch between current behaviour and the new way of dealing with permissions.

The PR is a draft because the base functionality for editing and checking is complete but not yet completely tested. Currently it works for users, groups and test types. The idea is to get feedback. Next steps will be unit tests and integration tests for the base functionality and including more configurations.

**Screenshots**

- Setting permissions for a user by a user who is allowed to change permissions:
  2021-11-11 07_26_41-View User _ DefectDojo

- Viewing permissions for a group by a user who is not allowed to change permissions:
  2021-11-11 07_27_54-View Group _ DefectDojo

### CODE REVIEW

File: dojo/api_v2/views.py [hunk 1]
```diff
@@ -106,6 +108,14 @@ class DojoGroupViewSet(prefetch.PrefetchListMixin,
...
         return get_authorized_groups(Permissions.Group_View).distinct()
 
+    def perform_create(self, serializer):
+        group = serializer.save()
+        group_post_create(group)
+
+    def perform_destroy(self, instance):
+        instance.delete()
+        group_post_delete(instance)
+
```
Comment (valentijnscholten) [inline dojo/api_v2/views.py:118]: Did you consider django signals as a way to implement post_create?

Comment (StefanFl) [reply dojo/api_v2/views.py:118]: Good idea, I will check

Comment (valentijnscholten) [reply dojo/api_v2/views.py:118]: Or overriding the `.save()` method

Comment (StefanFl) [reply dojo/api_v2/views.py]: changed to signals

**Comment by StefanFl** [comment]
Build for unit test fails with

```
#26 exporting cache
#26 preparing build cache for export 0.0s done
#26 writing layer sha256:0b7df1f26cfb993e08b866c38a90874fd041223ee9ccedee1b0caf3adee3f31b
#26 writing layer sha256:0b7df1f26cfb993e08b866c38a90874fd041223ee9ccedee1b0caf3adee3f31b 0.3s done
#26 writing layer sha256:36465807550f90174c68175fcc393ef439db0faf74421004f3e37f31ef4fbead
#26 writing layer sha256:36465807550f90174c68175fcc393ef439db0faf74421004f3e37f31ef4fbead 1.1s done
#26 writing layer sha256:4219e208f5fe4085f36dd7783d6946ce8e3eab2b6fc3fb640d678aa40cbf2294
#26 writing layer sha256:4219e208f5fe4085f36dd7783d6946ce8e3eab2b6fc3fb640d678aa40cbf2294 0.3s done
#26 writing layer sha256:43e9a3e494ae08e4855ca781d05cd7f63c11cf2508966ba5acbc8f357de6dd1b
#26 writing layer sha256:43e9a3e494ae08e4855ca781d05cd7f63c11cf2508966ba5acbc8f357de6dd1b 0.5s done
#26 writing layer sha256:5e9823d7f642ecbdfe2e305fa9ae3750cd3a0ee6f4ba80c03595b642045f901e 0.1s done
#26 ERROR: blob sha256:5e9823d7f642ecbdfe2e305fa9ae3750cd3a0ee6f4ba80c03595b642045f901e not found
------
 > exporting cache:
------
error: failed to solve: blob sha256:5e9823d7f642ecbdfe2e305fa9ae3750cd3a0ee6f4ba80c03595b642045f901e not found
Error: buildx failed with: error: failed to solve: blob sha256:5e9823d7f642ecbdfe2e305fa9ae3750cd3a0ee6f4ba80c03595b642045f901e not found
```

tried rebuilds 3 times, no idea what's happening here and couldn't find an answer with google

**Comment by valentijnscholten** [comment]
It's probably a bug in buildkit supposed to be fixed in their master branch. I'll see if I can get our tests to use that, or disable the caching for now.

**Comment by valentijnscholten** [comment]
If you rebase or merge `dev` it should start working again.

**Comment by StefanFl** [comment]
AccessLint isn't running anymore for this PR, after its change requests had been solved.

**Review by valentijnscholten — APPROVED** [review]

**Review by mtesauro — APPROVED** [review]
Looks good to me.  I assume Access Lint is just stuck rather then about to go red on us.

**Comment by StefanFl** [comment]
> I assume Access Lint is just stuck rather then about to go red on us.

Exactly, AccessLint initially had a few complaints which I resolved and AccessLint said: "You fixed the issue(s)! Great work." After that it never ran again.

**Pull request merged**